presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -542,6 +542,16 @@ public void dropPartition(String databaseName, String tableName, List<String> pa
     public void grantTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToGrant = privilegeGrantInfoSet.stream()
```

Comment:
I actually had named it earlier as `newPrivilegesToGrant`. But then thought that the "new" word here is kind of confusing. What we want is the set of requested privileges minus the privileges that user possesses. So couldn't come up with a good name that conveys this. So later I resorted to keeping it as `privilegesToGrant`.